### DISCUSSION TITLE
Recent Posts (Pages) Widget

### DISCUSSION THREAD

**Discussion by taufik-nurrohman** [mecha-cms/mecha#103] [Ideas]
Requires _page_ extension.

``` php
function recent_pages(string $path, int $take = 5) {
    $out = "";
    $pages = Pages::from(LOT . DS . 'page' . DS . $path);
    if ($pages->count) {
        $pages->sort([-1, 'time']);
        $pages->chunk($take, 0);
        $out .= '<ul>';
        foreach ($pages as $page) {
            $out .= '<li><a href="' . $page->url . '">' . $page->title . '</a></li>';
        }
        $out .= '</ul>';
    } else {
        $out .= '<p>No pages yet.</p>';
    }
    return $out;
}
```

### Usage

``` php
echo recent_pages('article', 5);
```

**Comment by yuliakbard** [comment]
kalo recent post nya muncul setiap postingan pada halaman home gimana kang? target path nya seperti apa?

**Reply by taufik-nurrohman** [reply]
`$path` itu mengarah ke folder penyimpanan laman. Bisa pakai target apa saja yang penting mengarah ke folder yang eksis.

**Reply by yuliakbard** [reply]
kalau ini kan ngambil pathnya ngambil folder article

`$pages = Pages::from(LOT . DS . 'page' . DS . $path);`

kalau ngambil path dari child article itu penulisannya seperti apa kang?

**Reply by taufik-nurrohman** [reply]
~~~ php
​echo​ ​recent_pages​(​'article/foo/bar/baz'​, ​5​);
~~~

**Reply by yuliakbard** [reply]
`echo​ ​recent_pages​(​'article/foo/bar/baz'​, ​5​);`

kang cara ngambil  path `article/foo/bar/baz` dari `$page->url` gimana ya kang? belum ketemu jawaban saya nih kang

**Reply by taufik-nurrohman** [reply]
Coba utak-atik pakai `URL::short()` (atau `short()` saja di Mecha 3.x.x).

**Reply by yuliakbard** [reply]
work kang, pake `URL::short()` ini, semalam ane pakenya yang `PATH::short()`. pantesan gak jalan, hehehe
